dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java

```diff
@@ -445,21 +457,25 @@ public void run() {
             }
             // find available processors
             int runtimeAvailableProcessors = Runtime.getRuntime().availableProcessors();
-            int fileSystemAvailableProcessors = getAvailableProcessorsFromFilesystem();
+            float fileSystemAvailableProcessors = getAvailableProcessorsFromFilesystemFloat();
 
-            int newAvailableProcessors;
-            if (fileSystemAvailableProcessors <= 0 || fileSystemAvailableProcessors > runtimeAvailableProcessors) {
-                newAvailableProcessors = runtimeAvailableProcessors;
-            } else {
-                newAvailableProcessors = fileSystemAvailableProcessors;
-            }
+            int currentNumberOfProcessorsInt = AVAILABLE_PROCESSORS_INTEGER.get();
+            float currentNumberOfProcessorsFloat = AVAILABLE_PROCESSORS_FLOAT.floatValue();
 
-            int currentNumberOfProcessors = AVAILABLE_PROCESSORS.get();
-            if (currentNumberOfProcessors != newAvailableProcessors) {
-                if (AVAILABLE_PROCESSORS.compareAndSet(currentNumberOfProcessors, newAvailableProcessors)) {
-                    notifyListeners(newAvailableProcessors);
+            if (currentNumberOfProcessorsInt != runtimeAvailableProcessors || currentNumberOfProcessorsFloat != fileSystemAvailableProcessors) {
+                if (fileSystemAvailableProcessors <= 0 || fileSystemAvailableProcessors > runtimeAvailableProcessors) {
+                    if (AVAILABLE_PROCESSORS_INTEGER.compareAndSet(currentNumberOfProcessorsInt, runtimeAvailableProcessors)) {
+                        notifyListeners(runtimeAvailableProcessors);
+                    }
+                    AVAILABLE_PROCESSORS_FLOAT.set(runtimeAvailableProcessors * 100);
```

Comment:
AVAILABLE_PROCESSORS_FLOAT maybe should be set first so that if we update CPUCount to have a getAsFloat() method it is set so that notify has the updated float value.

Reply:
The new review comment I gave moved the setting of the float to before the setting of the integer, but you didn't update this one to set the float value first.